### PR TITLE
[Hack Update] 027-FHIRPoweredHealthcare - Updated version 3 of FHIR Powered Healthcare WTH content

### PR DESCRIPTION
**Pull request by rsliang**
This is the updated version 3 of FHIR Powered Healthcare WTH content.

### CODE REVIEW

File: 027-FHIRPoweredHealthcare/Coach/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+This hack will deploy the following Azure resources and OSS components to implement the hack's challenges:
+- Azure Health Data Services workspace (managed PaaS in various challenges)
+- FHIR service	(managed FHIR server in challenge 1)
+- DICOM service (managed DICOM server in challenge 7)
+- MedTech service (managed PaaS to ingest and convert IoT medical device data into FHIR in challenge 6)
+- FHIR Loader (OSS) Function App based event-driven pipeline (for ingesting FHIR data in Challenge 1)
+- Azure Function (managed Serverless solution to host FHIR Loader app in challenge 1)
+- FHIR Analytics Pipeline - FHIR to Synapse sync agent (OSS pipeline to move FHIR data in FHIR service to Azure Data Lake for analytics with Synapse in challenge 4)
+- Tool for Health Data Anonymization pipleine (ADF pipeline to export and anonymize FHIR data in challenge 5)
```
Comment (jordanbean-msft) [inline 027-FHIRPoweredHealthcare/Coach/README.md:60]: pipeline

File: 027-FHIRPoweredHealthcare/Student/Challenge07.md [hunk 2]
```diff
@@ -1,66 +1,80 @@
...
+<center><img src="../images/challenge07-architecture.png" width="550"></center>
 
-<center><img src="../images/challenge07-architecture.jpg" width="550"></center>
+## Description
 
-A Timer is used to trigger the Logic App which bulk exports data from FHIR and stores in a preset storage location. The Logic App loops on an adjustable 5 minute interval until Bulk Export finishes exporting all data from FHIR. Logic App runs Azure Data Factory which in triggers Azure Batch which performs the deidentification with the FHIR Tools for Anonymization. The deidentified data is stored in Azure Data Lake Gen 2 for further use. 
+You will deploy an instance of DICOM service in your Azure Health Data Service workspace, and configure it to ingest DICOM files for persistence in the DICOM service.  Its DICOMcast pipeline will synchronize image metadata to FHIR service (deployed in challenge 1), which will enable healthcare organizations to integrate clinical and imaging data.  DICOMcast expands the heath data use cases by supporting a longitudinal patient data and creating cohorts for medical studies, analytics, and machine learning.
```
Comment (jordanbean-msft) [inline 027-FHIRPoweredHealthcare/Student/Challenge07.md:14]: "health", not "heath"

File: 027-FHIRPoweredHealthcare/Coach/Solution02.md [hunk 3]
```diff
@@ -1,101 +1,23 @@
...
 
-**[FHIR Converter](https://github.com/microsoft/FHIR-Converter)** is an open source project that runs as a REST web service for converting health data from legacy formats to FHIR bundles.  Microsoft FHIR Converter currently supports HL7v2 and C-CDA to FHIR conversion.  It uses pre-installed **[Handlebars templates](https://handlebarsjs.com/)** to define data mapping for HL7v2 to FHIR and C-CDA to FHIR conversion.  It can be deploy separately or as part of the **[HL7 to FHIR Conversion](https://github.com/microsoft/health-architectures/tree/master/HL7Conversion#hl7tofhir-conversion)** pipeline.
+**Convert data using the $convert-data endpoint**
+- Test send **[sample HL7v2](https://github.com/microsoft/FHIR-Converter/tree/main/data/SampleData/Hl7v2)** and **[sample C-CDA](https://github.com/microsoft/FHIR-Converter/tree/main/data/SampleData/Ccda)** requests in the request body payload and make the appropriate $convert-data API calls to receive FHIR Bundle response.
+            ```
```
Comment (jordanbean-msft) [inline 027-FHIRPoweredHealthcare/Coach/Solution02.md:14]: why 3 backticks?

File: 027-FHIRPoweredHealthcare/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+1. Deploy FHIR service in Azure Health Data Services platform and persist generated synthetic FHIR data into it.
+2. Convert and Load HL7v2 messages and C-CDA clinical data into FHIR Service.
+3. Develop JavaScript SPA (Single Page App) to search and view FHIR patient data.
+4. Transform and explore FHIR data for secondary use analytics.
+5. Export and anonymize FHIR data for downstream processing.
+6. Ingest, transform and load medical IoT device data into FHIR using MedTech service.
+7. Ingest, search and retrieve imaging data persisted in the DICOM service.
+
+## Challenges
+<center><img src="./images/challenges_architecture.png" width="850"></center>
```
Comment (jordanbean-msft) [inline 027-FHIRPoweredHealthcare/README.md:28]: is this image path correct?

**Review by jordanbean-msft — CHANGES_REQUESTED** [review]

File: 027-FHIRPoweredHealthcare/Student/Challenge07.md (hunk 2 above)
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Student/Challenge07.md]: Relaced 'heath' with 'health'  on line #14

File: 027-FHIRPoweredHealthcare/Coach/Solution02.md (hunk 3 above)
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution02.md]: Removed '```'

**Comment by rsliang** [comment]
I've pushed the requested changes above to my repo.  Do I need to submit another pull request?

**Comment by jordanbean-msft** [comment]
@microsoft/whatthehack-admins I have approved the content, need another approver

**Comment by jrzyshr** [comment]
This PR should close Issue #403

**Review by Whowong — CHANGES_REQUESTED** [review]

File: 027-FHIRPoweredHealthcare/Coach/Solution03.md [hunk 5]
```diff
@@ -1,98 +1,148 @@
...
+- Make sure the following Node.js prerequistes have been completed
+  - To see if you already have Node.js and npm installed and check the installed version, run: 
+    ```bash
+    node -v
+    npm -v
+    ```
+
+  - Download and install **[Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)**
+    - Download latest **[Node.js pre-built installer](https://nodejs.org/en/download/)** for your platform
+    - Run downloaded `node-v14.15.1-x64.msi` executable to install node.js
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution03.md]: Updated to 'Run installer downloaded..."

File: 027-FHIRPoweredHealthcare/Coach/Solution06.md [hunk 6]
```diff
@@ -1,148 +1,136 @@
...
-      - Add a submit button to perform the patient search
-
-  - **Option 2: Create React AAD MSAL Patient Search SPA**
-    - Build a new SPA in React using **[Create React App](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app)** toolchain.  It sets up your development environment with latest JavaScript features and optimizes your app for production. 
+In this challenge, you will work with IoT medical device data using the **[MedTech service](https://docs.microsoft.com/en-us/azure/healthcare-apis/iot/get-started-with-iot)** in Azure Health Data Services.  You will use **[MedTech service toolkit](https://github.com/microsoft/iomt-fhir/tree/main/tools/data-mapper)** to transform IoT medical device data into Fast Healthcare Interoperability Resources (FHIR®)-based Observation resources.  You will deploy a **[MedTech service data pipeline](microsoft.com/en-us/azure/healthcare-apis/iot/iot-data-flow)** to ingest medical IoT data, normaize and group these messages,transform the grouped-normalized messages into FHIR-based Observation resources, and then persist the transformed messages into the FHIR service (previously deployed in challenge 1).
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution06.md]: Fixed typo

File: 027-FHIRPoweredHealthcare/Coach/Solution06.md [hunk 7]
```diff
@@ -1,148 +1,136 @@
...
+                    - Replace <QUEUE NAME> with the name of the queue
+                - on `Header` tab, add the following
+                    - Add `Authorization` key and value: `Bearer <TOKEN from Azure AD>`
+                    Hint: Don't copy the enclosing double quotes in the token value
+                    - Add `Content-Type` key and `application/atom+xml;type=entry;charset=utf-8` as the value for it.
+                - On the `Body` tab,
+                    - Select `raw` for the data type
+                    - Enter the test device data as the message for the body
+                - Select `Send` to send the message to the Event Hub queue for processing by the MedTech service
+                    - If sucessful, you'll see the status as Created with the code 201 as shown in the following image.
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution06.md]: Fixed typo

File: 027-FHIRPoweredHealthcare/Coach/Solution06.md [hunk 8]
```diff
@@ -1,148 +1,136 @@
...
+                - On the `Event Hub Namespace` Overview page in the Azure portal, you'll' see that the messages are posted to the queue in the `Incoming Messages` section.
+- Verify device data is saved in the FHIR service as Observation resource(s) using Postman
+    - Open Postman collection: `WTH FHIR` (imported in challenge 1)
+    - First, run `AuthorizeGetToken SetBearer` API request and confirm `WTH FHIR` environment is selected in the top-right `Environment` drop-down. 
+        - Click the Send button to pass the values in the Body to AD Tenant, get the bearer token back and assign it to variable `bearerToken`.
+    - Run `Get Observation` API request and click the `Send` button. This will return all patients' Observation resources in the Response body.
+    - Search for the test device data persisted in the FHIR service as Obervation resource, i.e. sample Vital Signs data for Heart Rate.
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution06.md]: Fixed typo

File: 027-FHIRPoweredHealthcare/Coach/Solution06.md [hunk 9]
```diff
@@ -1,148 +1,136 @@
...
+        - In `Event Hubs Namespace`, select `Access Control (IAM)` page
+            - Select `+Add` to add Role Assignment
+                - Select Azure Event Hubs Data Sender for `Role` dropdown
+                - Select `Azure AD user, group or principla` for `Assign access to` dropdown
+                - Select your application for the service principal (name of client app/service principle created above).
+                - Select `Save` to add the new role assignment
+- Send events to an event hub via Postman
+    - Open Postman and **[import Postman data](https://learning.postman.com/docs/getting-started/importing-and-exporting-data/)**: 
+        - In Postman, click Import.
+        - In your **[Student Resources folder for Postman](../Student/Resources/Postman)**, select **[Environment](../Student/Resources/Postman/WTH FHIR-IoT-MedTech.postman_environment.json)** and **[Collection](../Student/Resources/Postman/WTH FHIR-IoT-MedTech.postman_collection.json)** JSON files.
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution06.md]: Updated links for Post env and collection.

File: 027-FHIRPoweredHealthcare/Student/Challenge01.md [hunk 10]
```diff
@@ -1,74 +1,67 @@
...
+First you will deploy **[Azure Health Data Services workspace](https://docs.microsoft.com/en-us/azure/healthcare-apis/workspace-overview)** and **[deploy a FHIR service](https://docs.microsoft.com/en-us/azure/healthcare-apis/fhir/fhir-portal-quickstart)** within the workspace.
 
-      **NOTE:** The connection to Azure AD can be made using a different tenant domain than the one tied to your Azure subscription. If you don't have privileges to create app registrations, users, grant admin consent, etc. in your Azure AD tenant, you can create a new secondary tenant, which will just be used for demo identities, etc. 
+You will then implement the **[FHIR Bulk Loader](https://github.com/microsoft/fhir-loader)** Function App solution to to ingest and load Synthea generated FHIR patient data into the FHIR service in near real-time.
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Student/Challenge01.md]: Removed duplicate 'to'

File: 027-FHIRPoweredHealthcare/Student/Challenge06.md [hunk 11]
```diff
@@ -1,64 +1,57 @@
...
+[< Previous Challenge](./Challenge05.md) - **[Home](../README.md)** - [Next Challenge>](./Challenge07.md)
 
 ## Introduction
 
-In this challenge, you will create a new JavaScript Single Page App (SPA) integrated with Microsoft Authentication Library (MSAL) to connect, read and search for FHIR patient data.
+In this challenge, you will work with IoT medical device data using the **[MedTech service](https://docs.microsoft.com/en-us/azure/healthcare-apis/iot/get-started-with-iot)** in Azure Health Data Services.  You will use **[MedTech service toolkit](https://github.com/microsoft/iomt-fhir/tree/main/tools/data-mapper)** to transform IoT medical device data into Fast Healthcare Interoperability Resources (FHIR®)-based Observation resources.  You will deploy a **[MedTech service data pipeline](https://learn.microsoft.com/en-us/azure/healthcare-apis/iot/iot-data-flow)** to ingest medical IoT data, normaize and group these messages,transform the grouped-normalized messages into FHIR-based Observation resources, and then persist the transformed messages into the FHIR service (previously deployed in challenge 1).
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Student/Challenge06.md]: Fixed typo

File: 027-FHIRPoweredHealthcare/Coach/Solution06.md [hunk 12]
```diff
@@ -1,148 +1,136 @@
...
\ No newline at end of file
+        - Run `Send message to devicedata` API HTTP Request in the `WTH FHIR-IoT-MedTech` Postman collection:
+        - First, open `AuthorizeGetToken SetBearer` and confirm `WTH FHIR-IoT-MedTech_event hub namespace` environment is selected in the top-right `Environment` drop-down. 
+            - Click the Send button to pass the values in the Body to AD Tenant, get the bearer token back and assign it to variable bearerToken.
+        - Open `Send message to devicedata`
+            - On the `Body` tab,
+                - Select `raw` for the data type
+                - Enter the test device data as the message for the body
+            - Select `Send` to send the message to the Event Hub queue for processing by the MedTech service
+                - If sucessful, you'll see the status as Created with the code 201 as shown in the following image.
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution06.md]: Fixed typo

File: 027-FHIRPoweredHealthcare/Coach/Solution07.md [hunk 13]
```diff
@@ -1,16 +1,62 @@
...
+  - Obtain access token to connect with your DICOM service
+    - Call `POST AuthorizeGetToken` API call in `Conformance-as-Postman` collection to obtian the access token needed to access DICOM data
+  - Store DICOM instance with sample DICOM files
+    - Select corresponding POST `Store-single-instance (xxx.dcm)` in `Conformance-as-Postman` collection for each sample DICOM files (red-triangle.dcm, green-square.dcm and blue-circle.dcm)
+    - Select the appropriate .dcm file (downloaded previously) for each API call in the `Body` tab.
+    - For each sample .dcm file, Send appropriate `POST Store-single-instance...` call to populate your DICOM service with the three .dcm single instance files.
+  - Call `Search-for-xxx` API calls in `Conformance-as-Postman` collection to Search for DICOM studies
+  - Call `Retrieve-xxx` API calls in `Conformance-as-Postman` collection to Retrieve DICOM studies
+  - Check logs for changes in DICOM service via Change Feed
+    - Call `GET /changefeed` API call to retrive logs of all the changes that occur in DICOM service
```
[the remark this answers is not in the document]
Comment (rsliang) [reply 027-FHIRPoweredHealthcare/Coach/Solution07.md]: Fixed typo

**Comment by jrzyshr** [comment]
@rsliang In Challenge 00, you say the coach will provide the students with a Resources.zip file.  This should contain the contents of the /Student/Resources folder.  However, I don't see anywhere else in the student guide where you refer to these files?  Are they used by students?

**Review by jrzyshr — CHANGES_REQUESTED** [review]
Thanks for the updates Richard!   Glad to see "V3" of the FHIR hack!

**Review by jrzyshr — APPROVED** [review]

**Comment by jrzyshr** [comment]
This PR closes issue #403

**Pull request merged**